src/python/pants/task/target_restriction_mixins.py

```diff
@@ -49,7 +49,40 @@ class HasSkipOptionMixin:
 
   @property
   def skip_execution(self):
-    return self.get_options().skip
+    return self.get_options().skip or self.resolve_only()
+
+  def resolve_only(self):
+    # This flag is defined only on fmt, which is done in FmtTaskMixin.
+    # In v2, we expect to have a --only flag on both fmt and lint which will allow individual
+    # formatters to be selected.
+    #
+    # This is a hacky one-off implementation to help Twitter deal with the fact that they have a
+    # custom Goal called scalafix, which does the equivalent of `fmt --fmt-only=scalafix`, as it
+    # provides them with a forward-compatible way of migrating people off of their scalafix goal.
+    #
+    # When the v2 goal is renamed from fmt2 to fmt, this option should be moved to that Goal, and
+    # its implementation broadened to support all v2 formatters, as well as any v1 formatters we
+    # fancy while we keep them around.
+    if hasattr(self.get_options(), "only"):
+      only = self.get_options().only
+      if only is None:
+        return False
+      elif only == "scalafix":
+        return not self.__class__.__name__.startswith("ScalaFix")
+      else:
+        raise ValueError("Invalid value for flag --only - must be scalafix or not set at all")
+    return False
+
+  def resolve_conflicting_skip_options(self, old_scope, new_scope, subsystem):
```

Comment:
Great idea!

Can you add type hints please?
```suggestion
  def resolve_conflicting_skip_options(self, old_scope: str, new_scope: str, subsystem: Subsystem):
```

Reply:
Done